Clash starter project doesn't compile on Windows

Running `stack build` yields an error:

```
bin\Clashi.hs:1:1: error:
    Could not find module `Prelude'
    Perhaps you haven't installed the "dyn" libraries for package `base-4.13.0.0'?
    Use -v (or :set -v in ghci) to see a list of the files searched for.
  |
1 | import Prelude
  | ^^^^^^^^^^^^^^

bin\Clashi.hs:2:1: error:
    Could not find module `System.Environment'
    Perhaps you haven't installed the "dyn" libraries for package `base-4.13.0.0'?
    Use -v (or :set -v in ghci) to see a list of the files searched for.
  |
2 | import System.Environment (getArgs)
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

bin\Clashi.hs:3:1: error:
    Could not find module `Clash.Main'
    Perhaps you haven't installed the "dyn" libraries for package `clash-ghc-1.2.2'?
    Use -v (or :set -v in ghci) to see a list of the files searched for.
  |
3 | import Clash.Main (defaultMain)
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Completed 104 action(s).

--  While building package clash-example-project-0.1 using:
      C:\sr\setup-exe-cache\x86_64-windows\Cabal-simple_Z6RU0evB_3.0.1.0_ghc-8.8.3.exe --builddir=.stack-work\dist\29cc6475 build lib:clash-example-project exe:clash exe:clashi --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

We shouldn't try to use `-dynamic` when on Windows.

Thanks to @LPoppe for reporting!

--- a/clash-starters/clash-example-project/clash-example-project.cabal
+++ b/clash-starters/clash-example-project/clash-example-project.cabal
@@ -78,12 +78,14 @@ library
 executable clash
   main-is: bin/Clash.hs
   Build-Depends: base, clash-ghc, clash-example-project
-  ghc-options: -dynamic
+  if !os(Windows)
+    ghc-options: -dynamic
 
 -- Builds the executable 'clashi', with clash-example-project in scope
 executable clashi
   main-is: bin/Clashi.hs
-  ghc-options: -dynamic
+  if !os(Windows)
+    ghc-options: -dynamic
   build-depends: base, clash-ghc, clash-example-project
 
 test-suite test-library

